src/main/java/org/wildfly/security/sasl/anonymous/AnonymousSaslClient.java

```diff
@@ -86,6 +87,6 @@ protected AnonymousSaslClient(final String protocol, final String serverName, fi
                 negotiationComplete();
                 return b.toArray();
         }
-        throw new SaslException("Invalid state");
+        throw new IllegalStateException();
```

Comment:
Now we can use ``Assert.unreachableCode`` and/or ``Assert.impossibleSwitchCase`` for this case.